packages/medusa/src/commands/utils/get-migrations.js

```diff
@@ -154,3 +190,66 @@ export const getEnabledMigrations = (migrationDirs, isFlagEnabled) => {
     })
     .filter(Boolean)
 }
+
+export const getModuleMigrations = (configModule, isFlagEnabled) => {
+  const loadedModules = getInternalModules(configModule)
+
+  const allModules = []
+
+  for (const loadedModule of loadedModules) {
+    const mod = loadedModule.loadedModule
+
+    const isolatedMigrations = {}
+    const moduleMigrations = (mod.migrations ?? [])
+      .map((migrations) => {
+        const all = []
+        for (const migration of Object.values(migrations)) {
+          // TODO: revisit how Modules export their migration entrypoints up/down
+          if (["up", "down"].includes(migration.name)) {
+            isolatedMigrations[migration.name] = migration
+          } else if (
+            typeof migration.featureFlag === "undefined" ||
+            isFlagEnabled(migration.featureFlag)
+          ) {
+            all.push(migration)
+          }
+        }
+        return all
+      })
+      .flat()
+
+    allModules.push({
+      moduleDeclaration: loadedModule.moduleDeclaration,
+      models: mod.models ?? [],
+      internalMigrations: moduleMigrations,
+      externalMigrations: isolatedMigrations,
+    })
+  }
+
+  return allModules
+}
+
+export const getModuleSharedResources = (configModule, featureFlagsRouter) => {
+  const isFlagEnabled = (flag) =>
+    featureFlagsRouter && featureFlagsRouter.isFeatureEnabled(flag)
+
+  const loadedModules = getModuleMigrations(configModule, isFlagEnabled)
+
+  let migrations = []
+  let models = []
+
+  for (const mod of loadedModules) {
+    if (mod.moduleDeclaration.resources !== "shared") {
+      continue
+    }
+
+    migrations = migrations.concat(mod.internalMigrations)
```

Comment:
The default value for this property is []

Reply:
Why not just migrations ?

Reply:
just want to be explicit whether it is a migration that will execute in Core's DB or a isolated one  where the core will just trigger the Module to migrate it

Reply:
I don't mind to change these property names though. @olivermrbl wdyt?

Reply:
Tell me if I am wrong but a module don't mix both migrations scenario right? If that is right then in that case, i would argue that it is on the resource type that it will be decided right? But in the end the module just export the migrations

Reply:
Not at this point, but I don't see why not in the future the resources could be `mixed`, some partes of the entities would go to the core, but the core would also trigger a migration of a second database the module is handling by itself.

Reply:
In that case the core only get the migrations that need to be run by the core. So there will never be non internal migrations here. It would just be the core saying to the module to run its isolated migrations

Reply:
changed it to `migrations` and kept `externalMigrations` that will be used later